frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts

```diff
@@ -0,0 +1,270 @@
+      this.handleInitializeNewNewTableData(designData); // <---- be careful doing this! if you see bugs, it may be because this is not the intended place for this function
+      // this.handleUpdateDesignDataTableChanges(designData);
+    }
+    this.updateFactorHeaders(designData);
+  }
+
+  handleInitializeExistingTableData() {
+    console.log('#init preexisting');
+
+    this.equalWeightFlag = this.experimentInfo.conditions.every(
+      (condition) => condition.assignmentWeight === this.experimentInfo.conditions[0].assignmentWeight
+    );
+
+    const newTableData = this.experimentDesignStepperService.mergeExistingConditionsTableData(this.experimentInfo);
+    this.initializeForm(newTableData);
+  }
+
+  handleInitializeNewNewTableData(designData: ExperimentFactorialDesignData) {
+    console.log('#init new');
+    this.equalWeightFlag = true;
+    const newTableData = this.experimentDesignStepperService.createNewFactorialConditionTableData(designData);
+    this.initializeForm(newTableData);
+  }
+
+  // handleUpdateDesignDataTableChanges(designData: ExperimentFactorialDesignData) {
+  //   console.log('#update table');
+  //   // TODO: intelligently handle updates to design data without triggering complete table re-creation
+  // }
+
+  updateFactorHeaders(designData: ExperimentFactorialDesignData) {
+    this.factorHeaders = designData.factors.map((factor) => {
+      return factor.factor;
+    });
+  }
+
+  initializeForm(tableData: FactorialConditionTableRowData[]) {
+    this.createFormControls(tableData);
+    const newTableData = this.applyEqualWeights(tableData);
+    this.experimentDesignStepperService.updateFactorialTableData(newTableData);
+    this.formInitialized = true;
+  }
+
+  resetEdit(): void {
+    this.experimentDesignStepperService.clearFactorialConditionTableEditModeDetails();
+  }
+
+  applyEqualWeights(partiallyUpdatedTableData?: FactorialConditionTableRowData[]): FactorialConditionTableRowData[] {
+    const tableData = partiallyUpdatedTableData || this.getCurrentTableData();
+
+    if (this.equalWeightFlag) {
+      const includedConditionsCount = this.getIncludedConditionCount(tableData);
+      const equalWeight = 100 / includedConditionsCount;
+      const newTableData = this.setNewWeights(tableData, equalWeight);
+
+      return newTableData;
+    } else {
+      return tableData;
+    }
+  }
+
+  setNewWeights(tableData: FactorialConditionTableRowData[], equalWeight: number): FactorialConditionTableRowData[] {
+    const newTableData = this.getFactorialConditions().controls.map((control, index) => {
+      const thisRowWeight: number = control.get('include').value ? equalWeight : 0;
+      control.get('weight').setValue(thisRowWeight.toFixed(1));
+      return { ...tableData[index], weight: this.experimentDesignStepperService.formatDisplayWeight(thisRowWeight) };
+    });
+
+    return newTableData;
+  }
+
+  validateWeightSumEqualTo100(factorialConditions: any) {
```

Comment:
please replace `any` with proper types

Reply:
it is getting direct new form values, which don't have specific type so it's kept as any.

Reply:
it can definitely get a specific type, if there isn't one created we should create that type. we shouldn't be using `any` anywhere in new code without a really good reason.